src/main/java/com/google/devtools/build/lib/vfs/FileSystem.java

```diff
@@ -74,19 +74,30 @@ public NotASymlinkException(PathFragment path, Throwable cause) {
    * structured/parsed representation, it will not cause any IO. (e.g., it will not resolve symbolic
    * links if it's a Unix file system.
    */
-  public Path getPath(String path) {
+  public final Path getPath(String path) {
     return Path.create(path, this);
   }
 
   /** Returns an absolute path instance, given an absolute path fragment. */
-  public Path getPath(PathFragment pathFragment) {
+  public final Path getPath(PathFragment pathFragment) {
     return Path.create(pathFragment, this);
   }
 
   final Root getAbsoluteRoot() {
     return absoluteRoot;
   }
 
+  /**
+   * Returns the underlying file system if the current file system is a partial overlay of another
+   * file system. Otherwise, returns this.
+   *
+   * <p>Note that the returned file system may still be an in-memory file system (in tests, for
+   * example), but should be treated as the "native" file system for the host machine.
+   */
+  public FileSystem getUnderlyingFileSystem() {
```

Comment:
Is "underlying" the best descriptor we can use? For RemoteActionFileSystem I think this can get confusing, because it's not a overlay but more like a union of two filesystems (that could both be said to "underlie" it).

If we can't come up with a better term, let's at least include a paragraph along these lines in the Javadoc: "for an action filesystem, this should return the on-disk component (or its underlying filesystem, if it is itself an overlay).